page-tracking/src/page_info.rs

```diff
@@ -225,9 +231,9 @@ impl PageInfo {
                     .try_push(owner)
                     .map_err(|_| PageTrackingError::OwnerOverflow)?;
                 self.state = new_state;
+                self.unlock_page().unwrap();
```

Comment:
Instead of a separate `is_locked()` check + `unlock_page()` you could do `unlock_page()?` before pushing the new owner / changing state.